phpBB/phpbb/template/twig/extension/icon.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\template\twig\extension;
+
+use phpbb\template\twig\environment;
+
+class icon extends \Twig\Extension\AbstractExtension
+{
+	/** @var \phpbb\user */
+	protected $user;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param \phpbb\user	$user			User object
+	 */
+	public function __construct(\phpbb\user $user)
+	{
+		$this->user = $user;
+	}
+
+	/**
+	 * Returns the name of this extension.
+	 *
+	 * @return string						The extension name
+	 */
+	public function getName()
+	{
+		return 'icon';
+	}
+
+	/**
+	 * Returns a list of functions to add to the existing list.
+	 *
+	 * @return \Twig\TwigFunction[]			Array of twig functions
+	 */
+	public function getFunctions()
+	{
+		return [
+			new \Twig\TwigFunction('Icon', [$this, 'icon'], ['needs_environment' => true]),
+		];
+	}
+
+	/**
+	 * Generate icon HTML for use in the template, depending on the mode.
+	 *
+	 * @param environment	$environment	Twig environment object
+	 * @param string		$type			Icon type (font|png|svg)
+	 * @param string		$icon			Icon name (eg. "bold")
+	 * @param string		$title			Icon title
+	 * @param bool			$hidden			Hide the icon title from view
+	 * @param string		$classes		Additional classes (eg. "fa-fw")
+	 * @param array			$attributes		Additional attributes for the icon, where the key is the attribute.
+	 *                      				{'data-ajax': 'mark_forums'} results in ' data-ajax="mark_forums"'
+	 * @return string
+	 */
+	public function icon(environment $environment, $type, $icon, $title = '', $hidden = false, $classes = '', array $attributes = [])
+	{
+		$type = utf8_strtolower($type);
```

Comment:
Since the only relevant input is ASCII, you can use `strtolower()` here so you don't depend on the mbstring polyfill.